### ISSUE TITLE
panic when network is set to null in k0s config

### ISSUE THREAD

**Issue by twz123** [k0sproject/k0s#2133]
### Before creating an issue, make sure you've checked the following:

- [X] You are running the latest released version of k0s
- [X] Make sure you've searched for existing issues, both open and closed
- [X] Make sure you've searched for PRs too, a fix might've been merged already
- [X] You're looking at docs for the released version, "main" branch docs are usually ahead of released versions.

### Platform

```shell
Linux 5.15.63-0-virt #1-Alpine SMP Mon, 29 Aug 2022 07:17:28 +0000 x86_64 GNU/Linux
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.16.2
PRETTY_NAME="Alpine Linux v3.16"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://gitlab.alpinelinux.org/alpine/aports/-/issues"
```

### Version

v1.24.2+k0s.0-204-gf61c9b44c

### Sysinfo

<details><summary>`k0s sysinfo`</summary>

```text
Machine ID: "39c2127b5dbb78bde7dad9280f39509b81e8699e9f993bb98125abff2c24f9d5" (from machine) (pass)
Total memory: 983.9 MiB (warning: 1.0 GiB recommended)
Disk space available for /var/lib/k0s: 6.8 GiB (pass)
Operating system: Linux (pass)
  Linux kernel release: 5.15.63-0-virt (pass)
  Max. file descriptors per process: current: 4096 / max: 4096 (warning: < 65536)
  Executable in path: modprobe: /sbin/modprobe (pass)
  /proc file system: mounted (0x9fa0) (pass)
  Control Groups: version 1 (pass)
    cgroup controller "cpu": available (pass)
    cgroup controller "cpuacct": available (pass)
    cgroup controller "cpuset": available (pass)
    cgroup controller "memory": available (pass)
    cgroup controller "devices": available (pass)
    cgroup controller "freezer": available (pass)
    cgroup controller "pids": available (pass)
    cgroup controller "hugetlb": available (pass)
    cgroup controller "blkio": available (pass)
  CONFIG_CGROUPS: Control Group support: no kernel config found (warning)
  CONFIG_NAMESPACES: Namespaces support: no kernel config found (warning)
  CONFIG_NET: Networking support: no kernel config found (warning)
  CONFIG_EXT4_FS: The Extended 4 (ext4) filesystem: no kernel config found (warning)
  CONFIG_PROC_FS: /proc file system support: no kernel config found (warning)
```
</details>

### What happened?

When setting `network: null` in k0s config, k0s segfaults.

### Steps to reproduce

1. Put an invalid config into `/etc/k0s/k0s.yaml`
    ```yaml
    apiVersion: k0s.k0sproject.io/v1beta1
    kind: ClusterConfig
    spec:
      api:
        address: 10.70.123.36
        sans:
        - 10.70.123.36
        - 127.0.0.1
      network: null # <- this breaks it
    ```
2. Run k0s. Any subcommand will fail, as far as I reckon.

### Expected behavior

No panic, clear error message. Probably even better: Identical behavior for `null` fields and absent fields.

### Actual behavior

```console
$ sudo k0s
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x11cab8e]

goroutine 1 [running]:
github.com/k0sproject/k0s/pkg/apis/k0s.k0sproject.io/v1beta1.(*Network).Validate(0x0)
        /go/src/github.com/k0sproject/k0s/pkg/apis/k0s.k0sproject.io/v1beta1/network.go:63 +0x2e
github.com/k0sproject/k0s/pkg/apis/k0s.k0sproject.io/v1beta1.validateSpecs(...)
        /go/src/github.com/k0sproject/k0s/pkg/apis/k0s.k0sproject.io/v1beta1/clusterconfig_types.go:356
github.com/k0sproject/k0s/pkg/apis/k0s.k0sproject.io/v1beta1.(*ClusterConfig).Validate(0xc000003180)
        /go/src/github.com/k0sproject/k0s/pkg/apis/k0s.k0sproject.io/v1beta1/clusterconfig_types.go:276 +0x330
github.com/k0sproject/k0s/pkg/config.(*ClientConfigLoadingRules).ParseRuntimeConfig(0xc000c1eee0)
        /go/src/github.com/k0sproject/k0s/pkg/config/file_config.go:134 +0x6bc
github.com/k0sproject/k0s/pkg/config.(*ClientConfigLoadingRules).readRuntimeConfig(...)
        /go/src/github.com/k0sproject/k0s/pkg/config/file_config.go:55
github.com/k0sproject/k0s/pkg/config.(*ClientConfigLoadingRules).Load(0x419a73?)
        /go/src/github.com/k0sproject/k0s/pkg/config/config.go:116 +0x6a
github.com/k0sproject/k0s/pkg/config.getClusterConfig({{0xc00005d7a0, 0x1b}, {0xc0008b8c10, 0x10}, {0xc0008b8c00, 0x10}, {0x30f8ec2, 0x12}, {0x30d995b, 0xc}, ...})
        /go/src/github.com/k0sproject/k0s/pkg/config/cli.go:295 +0xbd
github.com/k0sproject/k0s/pkg/config.GetCmdOpts()
        /go/src/github.com/k0sproject/k0s/pkg/config/cli.go:234 +0x218
github.com/k0sproject/k0s/cmd/ctr.setDefaultValues({0xc00009eff0, 0x5, 0xc0004baf00?})
        /go/src/github.com/k0sproject/k0s/cmd/ctr/ctr.go:51 +0x17b
github.com/k0sproject/k0s/cmd/ctr.NewCtrCommand()
        /go/src/github.com/k0sproject/k0s/cmd/ctr/ctr.go:31 +0x45
github.com/k0sproject/k0s/cmd.NewRootCmd()
        /go/src/github.com/k0sproject/k0s/cmd/root.go:86 +0x10f
github.com/k0sproject/k0s/cmd.Execute()
        /go/src/github.com/k0sproject/k0s/cmd/root.go:194 +0x19
main.main()
        /go/src/github.com/k0sproject/k0s/main.go:48 +0x22b
```

### Additional context

See #2134.

**Comment by soider** [comment]
fixed titles, because panic is different from segfault

**Comment by twz123** [comment]
(signal SIGSEGV is a segfault, it's just trapped by the Go runtime and converted to a panic.)

**Comment by makhov** [comment]
Fixed in #2136
